Running dsm-swap raises a bare `AssertionError`

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Steps to reproduce the problem

```python3
from qiskit.providers.fake_provider import FakeHanoiV2
from qiskit.circuit.library import QFT
backend = FakeHanoiV2()
qc = QFT(20)
qc.measure_all()
transpile(qc, backend, routing_method='dsm')
```
### What is the current behavior?

An `AssertionError` with no message is raised

### What is the expected behavior?

No error is raised

As written, the example currently raises for me with: 
```python
~/q/sketch/lib/python3.7/site-packages/dsm_swap/qiskit.py in run(self, dag)
     75             dagd = dag_decompose(dag)
     76             layer_perms = rh_knitter(
---> 77                 dagd.get_adjs(mode="coo"),
     78                 max_optim_steps=self.max_optim_steps,
     79                 seed=self.seed,

~/q/sketch/lib/python3.7/site-packages/dsm_swap/utils.py in get_adjs(self, mode)
    204             m = np.zeros((n, n), dtype=int)
    205             for g in layer.op_nodes(include_directives=False):
--> 206                 assert g.op.num_clbits == 0
    207                 if g.op.num_qubits == 1:
    208                     continue

AssertionError: 
```

If I comment out `qc.measure_all()`, my python kernel dies with a seg fault.

--- a/dsm_swap/utils.py
+++ b/dsm_swap/utils.py
@@ -21,6 +21,7 @@ from qiskit.dagcircuit import DAGCircuit
 from qiskit.converters import circuit_to_dag
 from qiskit.circuit.library import SwapGate
 from qiskit.transpiler.layout import Layout
+from qiskit.circuit import Measure
 import numpy as np
 from typing import Collection, Union, Tuple, List
 
@@ -203,6 +204,8 @@ class DecomposedDAG:
             n = layer.num_qubits()
             m = np.zeros((n, n), dtype=int)
             for g in layer.op_nodes(include_directives=False):
+                if isinstance(g.op, Measure):
+                    continue
                 assert g.op.num_clbits == 0
                 if g.op.num_qubits == 1:
                     continue

